### PR TITLE
Feature: show only future events on partner page

### LINKED ISSUE #347: In the Join us form, the "A ring back" and "More information" checkboxes are non-tabbable.
## Acceptance Criteria

- [ ] When I access the "Join us" form via keyboard, I can tab to and select "A ring back" and "More information" checkboxes

### PR DESCRIPTION
**Pull request by ivan-kocienski-gfsc**
Fixes #347

On partner's page, only show events from this partner that are in the future

### CODE REVIEW

**Comment by erbridge** [comment]
Which issue is this fixing? Looks like the linked one is related to "Join us".

**Comment by erbridge** [comment]
Ok. Found it (it's #346). That issue description says:

> If a partner has more than 20 events, they will be paginated with the weekly filter. If there are less than 20, display the "All events" and "Future events" filter buttons. Default to filter.

Can we do that full change, please so we don't block visibility of past events for small venues (where past events are even more important to show that they're real and active).

**Comment by ivan-kocienski-gfsc** [comment]
So in a conversation with @katjam we decided it would be easier to have the logic where if a partner has less than 20 events there's a list of "Upcoming Events" and below that there is a list of "Previous Events" (these sections are visible only if there are actual entries).

File: src/Page/Partners/Partner_.elm [hunk 1]
```diff
@@ -286,6 +275,52 @@ viewInfo localModel { partner, events } =
         ]
 
 
+viewPartnerEvents : Model -> Data -> Html Msg
+viewPartnerEvents localModel { partner, events } =
+    let
+        eventAreaTitle =
```
Comment (katjam) [inline src/Page/Partners/Partner_.elm:281]: It's a little confusing having this in a let block and line 309 inline.

If you wanted to factor out more of the mark-up generation, we could split section title into a function

File: src/Page/Partners/Partner_.elm [hunk 2]
```diff
@@ -286,6 +275,52 @@ viewInfo localModel { partner, events } =
...
+            h3 [ css [ smallInlineTitleStyle, color white ] ] [ text (t (PartnerUpcomingEventsText partner.name)) ]
+    in
+    section [ id "events" ]
+        (if List.length events > 0 then
```
Comment (katjam) [inline src/Page/Partners/Partner_.elm:285]: It feels a little nested - logic if / else-y and not that easy to follow.

I think there is a simplification we could do here, but if you don't have time / inclination to refactor, I'm happy to accept the tech debt at this point.

**Review by katjam — COMMENTED** [review]
I'm happy for this to go in as it is, in the interest of time since we are about to launch.

Equally if you feel like refactoring over the weekend - or want me to, we can do that - as an exception to our usual working practices. :)

**Comment by katjam** [comment]
> So in a conversation with @katjam we decided it would be easier to have the logic where if a partner has less than 20 events there's a list of "Upcoming Events" and below that there is a list of "Previous Events" (these sections are visible only if there are actual entries).

Yes - we kind of decided this was the path or least resistance - for now.

**Review by katjam — APPROVED** [review]
Let's merge as it is.

**Comment by katjam** [comment]
Hey @ivan-kocienski-gfsc - as with the other, I am merging for you in the interest of time.

**Pull request merged**